pkg/webservice/state_dump.go

```diff
@@ -0,0 +1,207 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package webservice
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/apache/incubator-yunikorn-core/pkg/log"
+	metrics2 "github.com/apache/incubator-yunikorn-core/pkg/metrics"
+	"github.com/apache/incubator-yunikorn-core/pkg/scheduler"
+	"github.com/apache/incubator-yunikorn-core/pkg/webservice/dao"
+
+	"github.com/gorilla/mux"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultStateDumpPeriod = 60
+	stateDumpFilePath      = "yunikorn-state.txt"
+)
+
+var (
+	periodicStateDump bool
+	abort             chan struct{}
+	startStop         sync.Mutex
+	stateDump         sync.Mutex // guards against simultaneous periodic vs web request
+)
+
+type AggregatedStateInfo struct {
+	Timestamp          string
+	Partitions         []*dao.PartitionInfo
+	Applications       []*dao.ApplicationDAOInfo
+	AppHistory         []*dao.ApplicationHistoryDAOInfo
+	Nodes              []*dao.NodesDAOInfo
+	NodesUtilization   []*dao.NodesUtilDAOInfo
+	ClusterInfo        []*dao.ClusterDAOInfo
+	ClusterUtilization []*dao.ClustersUtilDAOInfo
+	ContainerHistory   []*dao.ContainerHistoryDAOInfo
+	Queues             []*dao.PartitionDAOInfo
+	SchedulerHealth    *dao.SchedulerHealthDAOInfo
```

Comment:
I am not sure if the health status should be included or not. Now it is set as liveness probe, so when the scheduler becomes unhealthy, it is logged and also it will be restarted.

Reply:
Makes sense, removing it.